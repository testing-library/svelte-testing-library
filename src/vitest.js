import { afterEach } from 'vitest'

import { act, cleanup } from './pure.js'

afterEach(async () => {
  await act()
  cleanup()
})
