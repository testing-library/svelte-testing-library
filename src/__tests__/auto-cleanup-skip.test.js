import { beforeAll, describe, expect, test } from 'vitest'

import Comp from './fixtures/Comp.svelte'

describe('auto-cleanup-skip', () => {
  let render

  beforeAll(async () => {
    process.env.STL_SKIP_AUTO_CLEANUP = 'true'
    const stl = await import('..')
    render = stl.render
  })

  // This one verifies that if STL_SKIP_AUTO_CLEANUP is set
  // then we DON'T auto-wire up the afterEach for folks
  test('first', () => {
    render(Comp, { props: { name: 'world' } })
  })

  test('second', () => {
    expect(document.body.innerHTML).toMatchSnapshot()
  })
})
