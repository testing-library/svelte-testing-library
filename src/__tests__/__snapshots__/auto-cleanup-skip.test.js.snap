// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`auto-cleanup-skip > second 1`] = `""`;
